Ignore or address Flake8 gripes

Flake8 is not happy with certain Python stylings. Address those.

- [ ] cubedash/summary/_model.py:243:13: B028 No explicit stacklevel keyword argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.
- [ ] cubedash/summary/_schema.py:262:9: B028 No explicit stacklevel keyword argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.
- [ ] cubedash/summary/_schema.py:409:9: B028 No explicit stacklevel keyword argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.
- [ ] cubedash/summary/_stores.py:1900:21: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
- [ ] cubedash/summary/_stores.py:1904:25: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
- [ ] cubedash/summary/_stores.py:1904:34: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
- [ ] cubedash/summary/_stores.py:1904:44: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
- [ ] cubedash/summary/_stores.py:1904:57: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.
- [ ] cubedash/summary/_stores.py:1908:46: B031 Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage. Save the result to a list, if the result is needed multiple times.

--- a/cubedash/summary/_model.py
+++ b/cubedash/summary/_model.py
@@ -240,7 +240,7 @@ class TimePeriodOverview:
         if not self.footprint_geometry:
             return None
         if not self.footprint_crs:
-            warnings.warn(f"Geometry without a crs for {self}")
+            warnings.warn(f"Geometry without a crs for {self}", stacklevel=2)
             return None
 
         return (

--- a/cubedash/summary/_schema.py
+++ b/cubedash/summary/_schema.py
@@ -263,7 +263,8 @@ def is_compatible_schema(engine: Engine) -> bool:
             "Your database has item `cubedash.mv_region` from an unstable version of Explorer. "
             "It will not harm you, but feel free to drop it once all Explorer instances "
             "have been upgraded: "
-            "    drop materialised view cubedash.mv_region"
+            "    drop materialised view cubedash.mv_region",
+            stacklevel=2,
         )
 
     return is_latest
@@ -417,7 +418,8 @@ def check_or_update_odc_schema(engine: Engine):
 
                 {unexecuted_sql}
         """
-            )
+            ),
+            stacklevel=2,
         )
         raise
 

--- a/cubedash/summary/_stores.py
+++ b/cubedash/summary/_stores.py
@@ -1896,8 +1896,8 @@ def _common_paths_for_uris(
     def uri_scheme(uri: str):
         return uri.split(":", 1)[0]
 
-    for scheme, uris in groupby(sorted(uri_samples), uri_scheme):
-        uris = list(uris)
+    for scheme, uri_grouper in groupby(sorted(uri_samples), uri_scheme):
+        uris = list(uri_grouper)
 
         # Use the first, last and middle as examples
         # (they're sorted, so this shows diversity)

